File: app/src/main/java/jp/kitoha/ninow/Data/Config/StatusInfo.java
package jp.kitoha.ninow.Data.Config;

import android.content.SharedPreferences;
import android.os.Environment;
import android.webkit.JavascriptInterface;

import java.io.File;
import java.io.FileOutputStream;
import java.io.IOException;
import java.io.InputStream;
import java.io.OutputStream;

import jp.kitoha.ninow.Common.Constants;
import jp.kitoha.ninow.Common.ErrorCode;
import jp.kitoha.ninow.Common.ScreenCode;

/******************************************************************************
 * @author  KITOHA    N.Endo
 * @brief   進捗情報
 * @note    進捗ステータス
 * @since 2015 -
 * @copyright (c)KITOHA.co., ltd All right reserved.
 ******************************************************************************/
public class StatusInfo {
    //region シングルトン用コード
    /**************************************************************************
     * インスタンス変数
     **************************************************************************/
    public static StatusInfo instance = new StatusInfo();

    /****************************************************************
     * メソッド
     ****************************************************************/
    /**
     * @brief	インスタンスの取得
     * @return	インスタンス
     */
    public static StatusInfo getInstance(){
        return instance;
    }
    //endregion

    //region データ設定
    /**************************************************************************
     * インスタンス変数
     **************************************************************************/
    //業務情報
    private int     progress;				            //進捗状況
    private int     current_screen_id;		            //現在表示画面
    private int     prev_screen_id;	    	            //直前表示画面
    //コンテキスト(ファイル操作で利用)
    private SharedPreferences     pref;

    /**
     * @brief コンストラクタ
     */
    private StatusInfo(){
        progress            = 1;
        current_screen_id   = 1001;
        prev_screen_id      = 0;
        //コンテキスト(ファイル操作で利用)
        pref                = null;
    }
    /**
     * 設定の読み込み
     * @return
     */
    @JavascriptInterface
    public int read(){
        this.progress           = pref.getInt( Constants.KEY_PROGRESS, 1 );
        this.current_screen_id  = pref.getInt( Constants.KEY_CURRENT_SCR_ID, ScreenCode.PAGENO_F01A );
        this.prev_screen_id     = pref.getInt( Constants.KEY_PREV_SCR_ID, ScreenCode.PAGENO_F01A );

        return ErrorCode.STS_OK;
    }

    /**
     * 設定の保存
     * @return
     */
    @JavascriptInterface
    public int save(){
        SharedPreferences.Editor    editor  = pref.edit();

        //iniファイル形式のデータを読取る(sectionは使わない)
        editor.putInt( Constants.KEY_PROGRESS,              this.progress );
        editor.putInt( Constants.KEY_CURRENT_SCR_ID,        this.current_screen_id );
        editor.putInt( Constants.KEY_PREV_SCR_ID,           this.prev_screen_id );

        editor.commit();

        return ErrorCode.STS_OK;
    }
    //endregion

    //region 進捗状況
    @JavascriptInterface
    public int getProgress() {
        return progress;
    }

    @JavascriptInterface
    public void setProgress(int progress) {
        this.progress = progress;
    }
    //endregion

    //region 表示画面情報
    @JavascriptInterface
    public int getCurrentScreenId() {
        return current_screen_id;
    }

    @JavascriptInterface
    public void setCurrentScreenId(int current_screen_id) {
        this.current_screen_id = current_screen_id;
    }

    @JavascriptInterface
    public int getPrevScreenId() {
        return prev_screen_id;
    }

    @JavascriptInterface
    public void setPrevScreenId(int prev_screen_id) {
        this.prev_screen_id = prev_screen_id;
    }
    //endregion

    //region コンテキスト
    public SharedPreferences getPreferences() {
        return pref;
    }

    public void setPreferences(SharedPreferences pref) {
        this.pref = pref;
    }
    //endregion
}

File: app/src/main/assets/html/js/common/data/status_info.js
/*******************************************************************************
 * 定数・変数
 *******************************************************************************/

/*******************************************************************************
 * メソッド
 *******************************************************************************/
var StatusInfo = function() {
	var progress			= 1;				//進捗状況
	var current_screen_id	= 1001;				//現在表示画面
	var prev_screen_id		= 1001;				//前回表示画面
	return {
		//進捗状況の取得
		getProgress: function(){
			return progress;
		},
		setProgress: function(val){
			progress	= val;
		},
		//現在表示画面ID
		getCurrentScreenId: function(){
			return current_screen_id;
		},
		setCurrentScreenId: function(val){
			current_screen_id	= val;
		},
		//前回表示画面ID
		getPrevScreenId: function(){
			return prev_screen_id;
		},
		setPrevScreenId: function(val){
			prev_screen_id	= val;
		},
		/**
		 * 設定の読み取り
		 */
		read: function(){
			//OSによってデータの取得元を変える
			var agent		= navigator.userAgent;
			if( ( agent.search( /iPhone/ ) != -1 ) ||
				( agent.search( /iPad/ ) != -1 ) ||
				( agent.search( /iPod/ ) != -1 ) ){					//iOSの場合
			}else if( agent.search( /Android/ ) != -1 ){			//Androidの場合
				progress			= jv_status_info.getProgress();
				current_screen_id	= jv_status_info.getCurrentScreenId();
				prev_screen_id		= jv_status_info.getPrevScreenId();
			}else{
				//値がある場合のみ上書き(それ以外は初期値を使用)
				if( LocalStorage.getItem( "status_info_progress" ) ){
					progress			= LocalStorage.getItem( "status_info_progress" );
				}
				if( LocalStorage.getItem( "status_info_cur_scr_id" ) ){
					current_screen_id	= LocalStorage.getItem( "status_info_cur_scr_id" );
				}
				if( LocalStorage.getItem( "status_info_prev_scr_id" ) ){
					prev_screen_id		= LocalStorage.getItem( "status_info_prev_scr_id" );
				}
			}
		},
		/**
		 * 設定保存
		 */
		save: function(){
			//OSによってデータの保存先を変える
			var agent		= navigator.userAgent;
			if( ( agent.search( /iPhone/ ) != -1 ) ||
				( agent.search( /iPad/ ) != -1 ) ||
				( agent.search( /iPod/ ) != -1 ) ){					//iOSの場合
			}else if( agent.search( /Android/ ) != -1 ){			//Androidの場合
				//アプリケーション情報の保存
				jv_status_info.setProgress( progress );
				jv_status_info.setCurrentScreenId( current_screen_id );
				jv_status_info.setPrevScreenId( prev_screen_id );
				jv_status_info.save();
			}else{
				LocalStorage.setItem( "status_info_progress",		progress );
				LocalStorage.setItem( "status_info_cur_scr_id",		current_screen_id );
				LocalStorage.setItem( "status_info_prev_scr_id",	prev_screen_id );
			}
		}
	}
}();
